crates/services/sync/src/import.rs

```diff
@@ -232,67 +224,56 @@ where
         .in_current_span()
         .await
     }
+}
 
-    async fn get_block_for_header(
-        result: anyhow::Result<SourcePeer<SealedBlockHeader>>,
-        p2p: Arc<P>,
-        consensus_port: Arc<C>,
-    ) -> anyhow::Result<Option<SealedBlock>> {
-        let header = match result {
-            Ok(h) => h,
-            Err(e) => return Err(e),
-        };
-        let SourcePeer {
-            peer_id,
-            data: header,
-        } = header;
-        let id = header.entity.id();
-        let block_id = SourcePeer { peer_id, data: id };
-
-        // Check the consensus is valid on this header.
-        if !consensus_port
-            .check_sealed_header(&header)
-            .trace_err("Failed to check consensus on header")?
-        {
-            tracing::warn!("Header {:?} failed consensus check", header);
-            return Ok(None)
+fn get_block_stream<
+    P: PeerToPeerPort + Send + Sync + 'static,
+    C: ConsensusPort + Send + Sync + 'static,
+>(
+    range: RangeInclusive<u32>,
+    params: &Config,
+    p2p: Arc<P>,
+    consensus: Arc<C>,
+) -> impl Stream<Item = impl Future<Output = anyhow::Result<Option<SealedBlock>>>> {
+    get_header_stream(range, params, p2p.clone()).map({
+        let p2p = p2p.clone();
+        let consensus_port = consensus.clone();
+        move |batch| {
+            {
+                let p2p = p2p.clone();
+                let consensus_port = consensus_port.clone();
+                tokio::spawn(async move {
```

Comment:
It would be nice to support [graceful shutdown](https://tokio.rs/tokio/topics/shutdown) for all spawned tasks.

You can also wait for the shutdown of all tasks here:
<img width="1031" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/5275a8e5-898e-40b5-a6c4-56db049d5af4">

Reply:
Agreed. I will look into creating a `RunnableTask` for the import tasks.